www/docs/configuration/providers.md

```diff
@@ -20,42 +29,73 @@ NextAuth.js is designed to work with any OAuth service, it supports OAuth 1.0, 1
+|         id          |                    Unique ID for the provider                    |             `string`              |   Yes    |
+|        name         |                Descriptive name for the provider                 |             `string`              |   Yes    |
+|        type         |              Type of provider, in this case `oauth`              |              `oauth`              |   Yes    |
+|       version       |            OAuth version (e.g. '1.0', '1.0a', '2.0')             |             `string`              |   Yes    |
+|        scope        |          OAuth access scopes (expects array or string)           |      `string` or `string[]`       |   Yes    |
+|       params        |             Additional authorization URL parameters              |     `{ grant_type: string }`      |   Yes    |
+|   accessTokenUrl    |               Endpoint to retrieve an access token               |             `string`              |   Yes    |
+|  authorizationUrl   |         Endpoint to request authorization from the user          |             `string`              |   Yes    |
+|   requestTokenUrl   |               Endpoint to retrieve a request token               |             `string`              |   Yes    |
+|     profileUrl      |             Endpoint to retrieve the user's profile              |             `string`              |   Yes    |
+|      clientId       |                 Client ID of the OAuth provider                  |             `string`              |   Yes    |
+|    clientSecret     |               Client Secret of the OAuth provider                |             `string`              |   Yes    |
+|       profile       |       A callback returning an object with the user's info        |   `(profile, tokens) => Object`   |   Yes    |
+|     protection      | Additional security for OAuth login flows (defaults to `state`)  | `[pkce]`,`[state]`,`[pkce,state]` |    No    |
+|        state        | Same as `protection: "state"`. Being deprecated, use protection. |             `boolean`             |    No    |
+|       headers       |      Any headers that should be sent to the OAuth provider       |             `Object`              |    No    |
+| authorizationParams |    Additional params to be sent to the authorization endpoint    |             `Object`              |    No    |
+|       idToken       |   Set to `true` for services that use ID Tokens (e.g. OpenID)    |             `boolean`             |    No    |
+|       region        |                    Only when using BattleNet                     |             `string`              |    No    |
+|       domain        |                Only when using certain Providers                 |             `string`              |    No    |
+|      tenantId       |     Only when using Azure, Active Directory, B2C, FusionAuth     |             `string`              |    No    |
+
```

Comment:
```suggestion
:::tip
Even if you are using a built-in provider, you can override any of these options to tweak the default configuration.

```js title=[...nextauth].js
import Providers from "next-auth/providers"
...
Providers.Auth0({
  clientId: process.env.CLIENT_ID,
  clientSecret: process.env.CLIENT_SECRET,
  domain: process.env.DOMAIN,
  scope: "openid your_custom_scope", // We do provide a default, but this will override it if defined
  profile(profile) {
    return {} // Return the profile in a shape that is different from the built-in one.
  }
})
...
:::
```